src/Octokit.Webhooks/Extensions/StringEnum.cs

```diff
@@ -1,21 +1,32 @@
 namespace Octokit.Webhooks.Extensions;
 
 using System;
+using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
 using System.Linq;
 using System.Runtime.Serialization;
 using JetBrains.Annotations;
 
 [PublicAPI]
-public sealed record StringEnum<TEnum>
+public readonly struct StringEnum<TEnum> : IEquatable<StringEnum<TEnum>>
```

Comment:
Was trivial to fix, but this change caused my code to stop compiling as it can't be null anymore so probably should have been v4:

```diff
- message.PullRequest.State?.Value != Octokit.Webhooks.Models.PullRequestEvent.PullRequestState.Open ||
+ message.PullRequest.State.Value != Octokit.Webhooks.Models.PullRequestEvent.PullRequestState.Open ||
```

Reply:
FYI this was also mentioned by @nickfloyd on a specific case in https://github.com/octokit/webhooks.net/pull/746#pullrequestreview-3177198587

Reply:
Sorry about that. Would you like me to unlist `3.1.0` and publish it again as `4.0.0`?

Reply:
I don't mind either way - was more just an FYI 🙂

Reply:
@JamieMagee What do you think about the null vs default enum change?

> From the perspective of this change a follow through should have been to recognize that changing from a record class to struct requires to take the previous defaults into account. Previously the standard assignment on the property was null with a suppress and the default value of deserialization quite likely null too.

Reply:
@danielmarbach If I understand you correctly, you're suggesting:
1. Using `StringEnum<TEnum>?` everwhere
2. Releasing a new major version `v4.0.0`

Reply:
@JamieMagee I was only trying to understand if you were aware of that change in behavior and OK with it. I don't have enough history to judge whether that is an acceptable change in a minor not whether that behavior change is diserable in a major. I can only speculate from my own experience but that might not be aligned with the expectation this libs and the target audience has